src/components/Confirm/Confirm.js

```diff
@@ -112,9 +112,11 @@ class Confirm extends Component {
   }
 
   onApiSuccess = (apiResponse) => {
-    const { method, nextStep, actions } = this.props
+    const { method, nextStep, actions, triggerOnUpload = () => {} } = this.props
     const { capture } = this.state
 
+    triggerOnUpload(apiResponse);
```

Comment:
Hi @lf94, as what I'm understanding at this point, you want to save the `apiResponse`, which is a JSON of several fields with `applicant_id`, document `id` and the 2 hrefs with document `id`. I believe that:
- You have the full access to `applicant_id`
- You have the necessary document `id` emitted via [`onComplete` callback](https://github.com/onfido/onfido-sdk-ui/#oncomplete-function-optional)

Can you please explain more about your need for such a dedicated `onUpload` callback?

Reply:
Yes, it's possible to get all the information at some point in time using multiple requests. The `onUpload` resolves the following:

* No extra requests to get the information needed to download the documents uploaded to Onfido
* Specifically fires after upload of a single document

The `onComplete` callback "fires when both the document and face have been successfully captured and uploaded". 
Waiting for a face upload makes sense when using `onComplete`, but it doesn't make sense when a developer wants to capture documents as they are uploaded.

Reply:
As it's been some time, we've decided to use the webhook instead, so this callback is no longer required for us, but it may be for someone else.

Reply:
> we've decided to use the webhook instead, so this callback is no longer required for us

I'm glad that you've figured out an alternative way to make this work instead of self-hosting a patched version of Onfido Web SDK.

> but it may be for someone else. I can image such a feature can be used for UX.

Yes, we always appreciate the contributions towards improving UX for the Web SDK. But IMHO this feature will involve more than just UX because it relates to the Web SDK design principles, which focus on providing a consistent & robust experience in media capturing for the end-users, not media retrieving for the integrators. I believe that the Web SDK should be a black box, receiving minimum inputs (the SDK token) and returning minimum outputs (the document & biometric IDs).

But I'll pass your idea to our team's discussions to see if we can fit it into our product roadmap.

Thank you for your contribution!

Reply:
>  I believe that the Web SDK should be a black box

If that's one of the goals then I totally agree with your assessment and also believe the feature is "too low level". :slightly_smiling_face: 

Thanks again Ethan.